### ISSUE TITLE
Pip install mannrs doesnt seem to work

### ISSUE THREAD

**Issue by ghost** [jaimeliew1/RustMann#12]
Hey!

I just tried to pip install it, bit it gave the following error: 

```
ERROR: Could not find a version that satisfies the requirement mannrs (from versions: none)
ERROR: No matching distribution found for mannrs
```

**Comment by jaimeliew1** [comment]
Hi Marcus,

Thanks for reaching out. I'm always working to improve the installation process. Could you share which command(s) you used to install Mann.rs, as well as some details about your project setup and more of the error output? I just tested the pip installation process in the README, and it worked for me. Any additional details would be appreciated so we can get to the bottom of this.


Cheers,

Jaime

**Comment by ghost** [comment]
EDIT: did you just update the readme? I am pretty sure that it said you could pip install it a moment ago? Ohh well, Imma just do the git clone then :) 

Hey, I definitely didnt expect such a fast response! 

I am simply running the `pip install mannrs` command on Sophia, where I have a conda env and rust installed. 

The error it gives is: 
```
ERROR: Could not find a version that satisfies the requirement mannrs (from versions: none)
ERROR: No matching distribution found for mannrs

```

I could see that you have recently updated the package, so I kinda just assumed that something broke. But if it works for you, then I guess I must be the problem haha

Best,

Marcus

**Comment by jaimeliew1** [comment]
Yes I am in the process of updating the package to be more easily installable. Currently, I am working to make `pip install mannrs` work on all platforms, but it is not quite there yet. In the meantime, I recommend trying to install from the github repository. You can use:

```
git clone https://github.com/jaimeliew1/Mann.rs.git
cd Mann.rs
pip install .
```

**Comment by ghost** [comment]
Ahh, okok, I didnt see that you updated the readme. 

You had me questioning my own sanity for a second. I was certain that I saw you could use pip install mannrs haha!

But thanks, this works just fine! 

Have a nice weekend

**Comment by jaimeliew1** [comment]
Hi Marcus,

I have just uploaded the new package to Pypi, so `pip install mannrs` might actually work now (still in the process of testing it, but it looks good so far). Let me know how it goes.

**Comment by ghost** [comment]
I have already installed it with the git clone way, and that works perfectly. I can try and pip install it aswell later. Imma let you know if there is any problems

**Comment by May0048** [comment]
Hey!
After compiling, there is an extra "target" folder in the list. How can I run the example properly?

Best,

**Comment by jaimeliew1** [comment]
Hi May0048,

What operating system and Python version are you using? It is likely that you can just run `pip install mannrs` to install it fully and run the example python scripts. 

Otherwise, please provide more information about the error messages you are receiving and what you have done so far,

Cheers,

Jaime

**Comment by May0048** [comment]
Hello, I am using the Ubuntu system and I suspect it may be related to the GPU. I have recompiled the CPU version and there are no such errors.


Best regards,


---- Replied Message ----
| From | Jaime ***@***.***> |
| Date | 01/16/2025 17:35 |
| To | ***@***.***> |
| Cc | May0048 ***@***.***>,
Comment ***@***.***> |
| Subject | Re: [jaimeliew1/Mann.rs] Pip install mannrs doesnt seem to work (Issue #12) |

Hi May0048,

What operating system and Python version are you using? It is likely that you can just run pip install mannrs to install it fully and run the example python scripts.

Otherwise, please provide more information about the error messages you are receiving and what you have done so far,

Cheers,

Jaime
